docs/guides/migration/pip-to-project.md

```diff
@@ -0,0 +1,420 @@
+`requirements.txt` files, the `uv.lock` file can represent arbitrary groups of dependencies, so
+multiple files are not needed to lock development dependencies.
+
+To learn more, see the [lockfile](../../concepts/projects/layout.md#the-lockfile) documentation.
+
+## Importing requirements files
+
+The simpest way to import requirements is with `uv add`:
+
+```
+$ uv add -r requirements.in
+```
+
+However, there is some nuance to this transition. Notice we used the `requirements.in` file, which
+does not pin to exact versions of packages so uv will solve for new versions of these packages. You
+may want to continue using your previously locked versions from your `requirements.txt` so, when
+switching over to uv, none of your dependency versions change.
+
+The solution is to add your locked versions as _constraints_. uv supports using these on `add` to
+preserved locked versions:
+
+```
+$ uv add -r requirements.in -c requirements.txt
+```
+
+Your existing versions will be retained when producing a `uv.lock` file.
+
+### Importing platform-specific constraints
+
+If your platform-specific dependencies have been compiled into separate files, you can still
+transition to a universal lockfile. However, you cannot just use `-c` to specify constraints from
+your existing platform-specific `requirements.txt` files because they do not include markers
+describing the environment and will consequently conflict.
+
+To add the necessary markers, use `uv pip compile` to convert your existing files. For example,
+given the following:
+
+```text title="requirements-win.txt"
+colorama==0.4.6
+    # via tqdm
+tqdm==4.67.1
+    # via -r requirements.in
+```
+
+The markers can be added with:
+
+```console
+$ uv pip compile requirements.in -o requirements-win.txt --python-platform windows --no-strip-markers
+```
+
+Notice the resulting output includes a Windows marker on `colorama`:
+
+```text title="requirements-win.txt"
+colorama==0.4.6 ; sys_platform == 'win32'
+    # via tqdm
+tqdm==4.67.1
+    # via -r requirements.in
+```
+
+When using `-o`, uv will constrain the versions to match the existing output file, if it can.
+
+Markers can be added for other platforms by changing the `--python-platform` and `-o` values for
+each requirements file you need to import, e.g., to `linux` and `macos`.
+
+Once each `requirements.txt` file has been transformed, the dependencies can be imported to the
+`pyproject.toml` and `uv.lock` with `uv add`:
+
+```console
+$ uv add -r requirements.in -c requirements-win.txt -c requirements-linux.txt
+```
+
+### Importing development dependency files
+
+As discussed in the [development dependencies](#development-dependencies) section, it's common to
+have have groups of dependencies for development purposes.
+
+To import development dependencies, use the `--dev` flag during `uv add`:
+
+```console
+$ uv add -r requirements-dev.in -c requirements-dev.txt --dev
+```
```

Comment:
Interesting, is this smart enough to diff out the dev part from the common dependencies? (Given you previously showed that requirements-dev.in files tend to include requirements.in)

If that diffing works, does this require you to `uv add requirements.in` first, and then `uv add requirements-dev.in --dev`, or will you get the same result regardless?

Reply:
> (Given you previously showed that requirements-dev.in files tend to include requirements.in)

Uhhh that might be a problem. I will test. Good catch!